src/main/java/net/fabricmc/loom/api/processors/JarProcessor.java

```diff
@@ -22,16 +22,40 @@
  * SOFTWARE.
  */
 
-package net.fabricmc.loom.processors;
+package net.fabricmc.loom.api.processors;
 
-import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
 
 import org.gradle.api.Project;
+import org.objectweb.asm.commons.Remapper;
+
+import net.fabricmc.loom.task.RemapJarTask;
 
 public interface JarProcessor {
-	void setup(Project project);
+	/**
+	 * Checks if the processed file is up to date.
+	 * @param project The project
+	 * @param path The jar which was processed last
+	 * @return Is this processor up to date with this (potentially unprocessed) jar
+	 */
+	boolean isUpToDate(Project project, Path path);
```

Comment:
I've read this javadoc for 10 times and I still don't understand what this method accepts or returns